crypto/keys/codec.go

```diff
@@ -1,20 +1,22 @@
 package keys
 
 import (
-	amino "github.com/tendermint/go-amino"
 	cryptoAmino "github.com/tendermint/tendermint/crypto/encoding/amino"
 
+	"github.com/cosmos/cosmos-sdk/codec"
 	"github.com/cosmos/cosmos-sdk/crypto/keys/hd"
 )
 
-var cdc = amino.NewCodec()
+var cdc *codec.Codec
 
 func init() {
```

Comment:
I know this is the style in a bunch of the SDK, but can we avoid using `init` and package level variables? This is def an anti-pattern and makes our code less modular and reusable. Like this specifically could just be something like:

```
func RegisterTypes(cdc *codec.Codec) *codec.Codec {}
```

Reply:
Meh. I think this is a conversation of another PR... I tend to disagree specifically for the codec global variable. I don't see why this is problematic - modules need to define their own CDC for use in times when the cdc cannot be passed in... nor does it need to! because typically, modules will _only_ be decoding their own types, only in specific situations to modules need the global codec. I guess we could change this to a function `ModuleCdc` which simply retrieves the module codec but there shouldn't be a functional difference.

Reply:
Why are some codecs private and some are public? If they're all sealed and some of them are public, why not just have the codec that is created in `app.go` (the true global god codec) be available to all modules?

Reply:
Resolving this conversation as we agreed out of band that this is OK so long as exposed codecs are _sealed_.